exercises/views.py

```diff
@@ -104,9 +108,9 @@ def submit_autograding_exercise(request, exercise):
     if form.is_valid():
         answer = form.cleaned_data['answer']
 
-        print 'solution', metadata, answer
+        #print 'solution', metadata, answer
         grader_function = get_grader(exercise.tag)
-        is_solution = grader_function(metadata, answer)
+        (is_solution, error_msg) = grader_function(metadata, answer)
```

Comment:
A better API would be to allow the `grader_function` to return either a boolean. You can attempt to extract the two elements like you do and in case of a non-iterable exception you can take the return value as a boolean directly. This will make the change backwards-compatible and it will also limit the amount of changes you need to make to existing graders, limiting the surface area for bug introduction.